lib/experimental/new_embed.dart

```diff
@@ -199,6 +200,28 @@ class NewEmbed {
     _initModules().then((_) => _initNewEmbed());
   }
 
+  /// Initializes a listener for messages from the parent window. Allows this
+  /// embedded iframe to display and run arbitrary Dart code.
+  void _initHostListener() {
+    window.addEventListener('message', (dynamic event) {
+      var data = event.data;
+      if (data is! Map) {
+        // Ignore unexpected messages
+        return;
+      }
+
+      var type = data['type'];
+
+      if (type == 'sourceCode') {
+        var sourceCode = data['sourceCode'];
+        userCodeEditor.document.value = sourceCode;
+      }
+    });
+
+    // Send a ready message
+    window.parent.postMessage({'sender': 'frame', 'type': 'ready'}, '*');
```

Comment:
Would it be better to send the ready message after NewEmbed has completely set itself up (in a second `then` on line 200, for instance)?